# Introducción a NLTK

In [1]:
# Cargamos librería
import nltk

# Descargamos dataset
nltk.download('cess_esp')

[nltk_data] Downloading package cess_esp to /home/missael/nltk_data...
[nltk_data]   Unzipping corpora/cess_esp.zip.


True

En este caso, descargamos _cess esp_, que es un corpus en español con titulares de noticias 👀.
## Expresiones regulares

Primero analizaremos los datos usando **expresiones regulares**.

In [4]:
# Libreria para regular expressions
import re

# Cargamos el corpus
corpus = nltk.corpus.cess_esp.sents()
print(corpus[0])
print(len(corpus))

['El', 'grupo', 'estatal', 'Electricité_de_France', '-Fpa-', 'EDF', '-Fpt-', 'anunció', 'hoy', ',', 'jueves', ',', 'la', 'compra', 'del', '51_por_ciento', 'de', 'la', 'empresa', 'mexicana', 'Electricidad_Águila_de_Altamira', '-Fpa-', 'EAA', '-Fpt-', ',', 'creada', 'por', 'el', 'japonés', 'Mitsubishi_Corporation', 'para', 'poner_en_marcha', 'una', 'central', 'de', 'gas', 'de', '495', 'megavatios', '.']
6030


Podemos ver un ejemplo de titular y vemos que tenemos $6030$ titulares, además de que ya está tokenizado todo. Ahora vamos a hacer _flatten_ a los datos, es decir, en lugar de tener una lista de listas, vamos a pasar todos los tokens en una sola lista. 

In [5]:
flatten = [word for list in corpus for word in list]
print(len(flatten))

192686


¡Tenemos muchísimas palabras!🤯